lib/string.js

```diff
@@ -292,18 +292,56 @@ internals.String = class extends Any {
         });
     }
 
-    guid() {
+    guid(guidOptions) {
 
-        const regex = /^[A-F0-9]{8}(?:-?[A-F0-9]{4}){3}-?[A-F0-9]{12}$/i;
-        const regex2 = /^\{[A-F0-9]{8}(?:-?[A-F0-9]{4}){3}-?[A-F0-9]{12}\}$/i;
+        // 89AB on last group of 4
+        // Version on second group of 4
+
+        const brackets = {
+            '{': '}', '[': ']', '(': ')', '': ''
+        };
+
+        const uuids = {
+            'uuidv1': '1',
+            'uuidv2': '2',
+            'uuidv3': '3',
+            'uuidv4': '4',
+            'uuidv5': '5'
+        };
+
+        const regex = /^([\[{\(]?)([0-9A-F]{8})([:-]?)([0-9A-F]{4})([:-]?)([0-9A-F]{4})([:-]?)([0-9A-F]{4})([:-]?)([0-9A-F]{12})([\]}\)]?)$/i;
 
         return this._test('guid', undefined, function (value, state, options) {
 
-            if (regex.test(value) || regex2.test(value)) {
-                return value;
+            const results = regex.exec(value);
+            if (!results) {
+                return this.createError('string.guid', { value }, state, options);
+            }
+
+            // Matching braces
+            if (brackets[results[1]] !== results[11]) {
```

Comment:
I'm not sure if they are eluding me or what but I'm not seeing any tests for mismatched braces.